### ISSUE TITLE
split() function with empty string as delimiter could cause database hanging and massive resource consumption

### ISSUE THREAD

**Issue by Aurora-yzj** [memgraph/memgraph#1828]
**Memgraph version**
2.15.0

**Environment**
Operating system: Ubuntu 22.04
Architecture: AMD64
Connect method: any connect method
Driver: any driver (not docker)

**Describe the bug**
When using the split() function with an empty string as a delimiter, the database will hang and occupy massive memory space. 

**To Reproduce**
1. Run the following command to start memgraph:
```
 ./memgraph --bolt-address 127.0.0.1 --bolt-port 37687 --storage-properties-on-edges true --query-execution-timeout-sec=1
```
2. Run the query:
```
RETURN split('abc', '')
```
Memgraph will stuck and take lots of memory space (executing 30 seconds will lead to nearly 30GiB memory space occupation):
![2024-03-18_10-54](https://github.com/memgraph/memgraph/assets/24768784/9431f5da-a639-4a99-99f4-2f595e9de40e)

Similar to the previous bug I reported, although I have set a limitation to the execution time, this query will still cause Memgraph to hang forever and consume massive resources. This could cause DoS attacks.

**Expected behavior**
Normally finishing executing the query.

Thank you so much for your help!

**Verification Environment**
* Plain memgraph binary
* Architecture: x86

**Comment by katarinasupe** [comment]
This issue is duplicated -> https://github.com/memgraph/memgraph/issues/1828 https://github.com/memgraph/memgraph/issues/609

@Aurora-yzj if it's okay, please track progress on https://github.com/memgraph/memgraph/pull/1785

**Comment by Aurora-yzj** [comment]
> This issue is duplicated -> #1828 #609
> 
> @Aurora-yzj if it's okay, please track progress on #1785

Thanks! I'll track the progress with that issue ID.

**Comment by katarinasupe** [comment]
Hi @Aurora-yzj, the fix should be available under Memgraph 2.16. Let me know if that works for you. Thanks again for reporting the issue and making Memgraph a better product 🚀
